src/lib/crypto/snarky_tests/snarky_tests.ml

```diff
@@ -191,6 +191,88 @@ let generate_witness () =
 
   ()
 
+module As_prover_circuits = struct
+  let ( -- ) i j =
+    let rec aux n acc = if n < i then acc else aux (n - 1) (n :: acc) in
+    aux j []
+
+  let input_typ = Impl.Typ.tuple3 Impl.Field.typ Impl.Field.typ Impl.Field.typ
+
+  let return_typ = Impl.Typ.unit
+
+  let main as_prover vars
+      ((b1, b2, b3) : Impl.Field.t * Impl.Field.t * Impl.Field.t) () =
+    let abc = Impl.Field.(b1 + b2 + b3) in
+    if as_prover then
+      Impl.as_prover (fun _ ->
+          let f acc e =
+            let var = Snarky_backendless.Cvar.(Var e) in
```

Comment:
you should use`Snarky_backendless.Cvar.Unsafe.of_int e` instead. I think we need to add an interface to cvar to prevent direct construction (but deconstruction allowed). I think that would be a good change, maybe we can talk about that next standup.